public/app/features/dashboard-scene/sharing/ShareExportTab.tsx

```diff
@@ -43,46 +68,140 @@ export class ShareExportTab extends SceneObjectBase<ShareExportTabState> impleme
     });
   };
 
+  public onExportModeChange = (exportMode: ExportMode) => {
+    this.setState({
+      exportMode,
+    });
+
+    if (exportMode === ExportMode.Classic) {
+      this.setState({
+        isViewingYAML: false,
+      });
+    }
+  };
+
   public onViewJSON = () => {
     this.setState({
       isViewingJSON: !this.state.isViewingJSON,
     });
   };
 
+  public onViewYAML = () => {
+    this.setState({
+      isViewingYAML: !this.state.isViewingYAML,
+    });
+  };
+
   public getClipboardText() {
     return;
   }
 
   public getExportableDashboardJson = async (): Promise<{
-    json: Dashboard | DashboardJson | DashboardV2Spec | { error: unknown };
+    json: Dashboard | DashboardJson | DashboardV2Spec | ExportableResource | { error: unknown };
     hasLibraryPanels?: boolean;
+    initialSaveModelVersion: 'v1' | 'v2';
   }> => {
-    const { isSharingExternally } = this.state;
+    const { isSharingExternally, exportMode } = this.state;
 
     const scene = getDashboardSceneFor(this);
     const exportableDashboard = await scene.serializer.makeExportableExternally(scene);
+    const initialSaveModel = scene.getInitialSaveModel();
+    const initialSaveModelVersion = initialSaveModel && isDashboardV2Spec(initialSaveModel) ? 'v2' : 'v1';
     const origDashboard = scene.serializer.getSaveModel(scene);
     const exportable = isSharingExternally ? exportableDashboard : origDashboard;
+    const metadata = getMetadata(scene, Boolean(isSharingExternally));
+
+    if (isDashboardV2Spec(origDashboard) && 'elements' in exportable && initialSaveModelVersion === 'v2') {
+      this.setState({
+        exportMode: ExportMode.V2Resource,
+      });
 
-    if (isDashboardV2Spec(origDashboard)) {
       return {
-        json: exportable,
+        json: {
+          kind: 'Dashboard',
```

Comment:
@harisrozajac let's change the order to:

- apiVersion
- kind
- metadata
- spec
- status